tempesta_db/core/main.c

```diff
@@ -222,6 +225,72 @@ tdb_get_db(const char *path, int node)
 	return tdb_get(db);
 }
 
+/**
+ * Lookup and get a record if the record is found or create TDB entry to store
+ * @len bytes. If record exist then since we don't copy returned records,
+ * we have to lock the memory location where the record is placed and
+ * the user must call tdb_rec_put() when finish with the record.
+ *
+ * The caller must not call sleeping functions during work with the record.
+ * Typically there is only one large record per bucket, so the bucket lock
+ * is exactly the same as to lock the record. While there could be many
+ * small records in a bucket, so the caller should not perform long jobs
+ * with small records.
+ *
+ * @return pointer to record with acquired bucket lock if the record is
+ * found and create TDB entry without acquired locks otherwise.
+ *
+ * TODO #515 rework the function is lock-free way
```

Comment:
Should be "**in** lock-free" and use a dot at the end of sentences. Please be more accurate in comments.